pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,176 @@
+			caps := newLibvirtCapsTopology(
+				withCells(cellCount),
+				withCellMemory(memorySize, unit),
+				withCellPages(pageCount, pageAmount, unit),
+				withCellDistances(cellDistance),
+				withCellCpus(cpuCount),
+			)
+			topology := caps.Host.NUMA
+			expectedTopology := capabilitiesToTopology(caps)
+
+			Expect(expectedTopology.NumaCells).To(HaveLen(cellCount))
+			for i, cell := range topology.Cells.Cells {
+				Expect(expectedTopology.NumaCells[i].Memory).To(PointTo(matchMemory(cell.Memory)))
+				Expect(expectedTopology.NumaCells[i].Pages).To(ConsistOf(matchPages(cell.PageInfo)))
+				Expect(expectedTopology.NumaCells[i].Distances).To(ConsistOf(matchDistances(cell.Distances.Siblings)))
+				Expect(expectedTopology.NumaCells[i].Cpus).To(ConsistOf(matchCPUs(cell.CPUS.CPUs)))
+			}
+		},
+			Entry("Single cell, single page, single CPU per cell", 1, 1, 1),
+			Entry("Multiple cells, multiple pages, multiple CPUs per cell", 4, 3, 4),
+		)
+	})
+})
+
+type option func(caps *libvirtxml.Caps)
+
+func newLibvirtCapsTopology(opts ...option) *libvirtxml.Caps {
+	caps := &libvirtxml.Caps{
+		Host: libvirtxml.CapsHost{
+			NUMA: &libvirtxml.CapsHostNUMATopology{},
+		},
+	}
+
+	for _, f := range opts {
+		f(caps)
+	}
+
+	return caps
+}
+
+func withCells(count int) option {
+	return func(caps *libvirtxml.Caps) {
+		caps.Host.NUMA.Cells = &libvirtxml.CapsHostNUMACells{
+			Cells: make([]libvirtxml.CapsHostNUMACell, count),
+		}
+		for i := 0; i < count; i++ {
+			caps.Host.NUMA.Cells.Cells[i] = libvirtxml.CapsHostNUMACell{
+				ID: i,
+			}
+		}
+	}
+}
+
+func withCellMemory(size uint64, unit string) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].Memory = &libvirtxml.CapsHostNUMAMemory{
+				Unit: unit,
+				Size: size,
+			}
+		}
+	}
+}
+
+func withCellPages(pageCount int, count uint64, unit string) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].PageInfo = make([]libvirtxml.CapsHostNUMAPageInfo, pageCount)
+			for j := range pageCount {
+				caps.Host.NUMA.Cells.Cells[i].PageInfo[j] = libvirtxml.CapsHostNUMAPageInfo{
+					Unit:  unit,
+					Count: count,
+					// Similar growth to common hugepages (4KiB, 2MiB. 1GiB)
+					Size: 4 << (9 * j),
+				}
+			}
+		}
+	}
+}
+
+func withCellDistances(distance int) option {
+	return func(caps *libvirtxml.Caps) {
+		for i := range caps.Host.NUMA.Cells.Cells {
+			caps.Host.NUMA.Cells.Cells[i].Distances = &libvirtxml.CapsHostNUMADistances{
+				Siblings: make([]libvirtxml.CapsHostNUMASibling, len(caps.Host.NUMA.Cells.Cells)),
+			}
+			for j := range caps.Host.NUMA.Cells.Cells {
+				// Doesn't mimic standard NUMA behavior, no differing latency
+				caps.Host.NUMA.Cells.Cells[i].Distances.Siblings[j] = libvirtxml.CapsHostNUMASibling{
+					ID:    j,
+					Value: distance,
+				}
+			}
+		}
+	}
+}
```

Comment:
same here

Reply:
I agree. Also see the comment here https://github.com/kubevirt/kubevirt/pull/12652#discussion_r1743340868

Reply:
Done.